atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,179 @@
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        # platform = os.uname()[4]
+        if self.is_orchestrator():
+            self.log.warning("%s plugin set to run on orchestrator. Skipping", self.key)
+            return False
+        # TODO: skip if not in selected platform
+        if is_scratch_build():
+            self.log.info("Scratch build. Skipping")
+            return False
+        if not self.has_operator_manifest():
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
+            return False
+        return True
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+        if not self.should_run():
+            return
+        saved_image = self.workflow.exported_image_sequence[-1].get('path')
+        self.log.debug("Extracting data from %s", saved_image)
+        temp_dir = tempfile.mkdtemp()
+        manifests_zipfile_path = self.extract_manifests(saved_image, temp_dir)
+        return manifests_zipfile_path
+
+    def get_archived_manifests(self, members):
+        for tarinfo in members:
+            if tarinfo.name.startswith(os.path.join(MANIFESTS_DIR_NAME, '')):
+                yield tarinfo
+
+    def extract_manifests(self, image_archive_path, workdir):
+        """
+        Extract operator manifest files from image layers
+
+        :param image_archive_path: str, path to docker image archive tar file
+        :param workdir: str, path to work directory
+
+        :return: str, path to operator manifests zip file
+        """
+        if not tarfile.is_tarfile(image_archive_path):
+            raise RuntimeError('Provided image is not a tar archive')
+
+        try:
+            image_archive = tarfile.TarFile(image_archive_path)
+        except tarfile.ReadError as ex:
+            self.log.error("Could not read image archive: %s" % ex)
+            raise RuntimeError("Could not read image archive: %s" % ex)
+
+        try:
+            manifest_info = image_archive.getmember(IMAGE_MANIFEST_NAME)
+        except KeyError as ex:
+            raise RuntimeError('Image archive format error. Manifest file not found %s' % ex)
+
+        if not manifest_info.isfile():
+            raise
+
+        manifest_buffer = image_archive.extractfile(manifest_info)
+        manifest = json.load(manifest_buffer)
+
+        for image in manifest:
+            # We rely on the order of the array in manifests.json
+            for layer_path in image['Layers']:
+                try:
+                    layer_archive_info = image_archive.getmember(layer_path)
+                except KeyError as ex:
+                    raise RuntimeError('Image archive format error. '
+                                       'Layer archive not found for %s: %s' % (layer_path, ex))
+                if not layer_archive_info.isfile():
```

Comment:
this is actually directory

Reply:
The strings in the Layers list point to the tar files inside each directory.

Reply:
ahh sorry I am blind, you are right